gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -112,10 +112,16 @@ def self._check_final_ancestors(target, target_ancestors, source_method_names)
             ancestor.private_method_defined?(method_name) ||
             ancestor.protected_method_defined?(method_name)) &&
            final_method?(method_owner_and_name_to_key(ancestor, method_name))
-          raise(
-            "The method `#{method_name}` on #{ancestor} was declared as final and cannot be " +
-            (target == ancestor ? "redefined" : "overridden in #{target}")
-          )
+          caller_loc = T.must(caller_locations&.find {|l| !l.to_s.match?(%r{sorbet-runtime[^/]*/lib/types/private/methods/}) })
+          definition_file, definition_line = T::Private::Methods.signature_for_method(ancestor.instance_method(method_name)).method.source_location
+
+          error_message = "The method `#{method_name}` on #{ancestor} was declared as final and cannot be " +
+                          (target == ancestor ? "redefined" : "overriden in #{target}")
+          pretty_message = "#{error_message}\n" \
+                           "Made final here: #{definition_file}:#{definition_line}\n" \
+                           "Overriden here: #{caller_loc.path}:#{caller_loc.lineno}\n"
+
+          T::Configuration.sig_validation_error_handler(pretty_message, {})
```

Comment:
Can we add a test that we're calling `sig_validation_error_handler` in the same way we do for other `T::Configuration` handlers? For example we should assert that if you set the sig validation handler to something that doesn't raise, that the code overall that has a final violation doesn't raise either.